### DISCUSSION TITLE
After dist upgrade from Ubuntu 20.04 to Ubuntu 22.04, no routing to Internet

### DISCUSSION THREAD

**Discussion by T313C0mun1s7** [pivpn/pivpn#1628] [Support]
## In raising this issue, I confirm the following:

`{please fill the checkboxes, e.g: [X]}`

- [X] I have read the [documentation](https://docs.pivpn.io)
- [] Is it a feature request? please consider opening a [Discussion] (https://github.com/pivpn/pivpn/discussions/new?category=feature-requests)
- [X] Is this a support request? please consider copying the content of this document and using it to open a [Discussion](https://github.com/pivpn/pivpn/discussions/new?category=support)
- [X] I have read and understood the [PiVPN General Guidelines](https://github.com/pivpn/pivpn/#general-guidelines)
- [] The issue I am reporting can be *replicated*.  
- [] The issue I am reporting *is* directly related to the pivpn installer script.
- [X] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pivpn/pivpn/wiki/FAQ), [closed issues](https://github.com/pivpn/pivpn/issues?q=is%3Aissue+sort%3Aupdated-desc+is%3Aclosed), and [open issues](https://github.com/pivpn/pivpn/issues?q=is%3Aissue+sort%3Aupdated-desc+is%3Aopen)).

### Has your install failed? 
```
no
```

### Describe the issue

I had a working install on Ubuntu 20.04.4 LTS Pro on AWS EC2 Instance. No issues of any kind with it. I then performed a dist upgrade to Ubuntu 22.04 LTS and rebooted. From then on I can connect to the VPN, but I don't route to the Internet. I did all of the trouble shooting and could not find an issue. So I then ran `pivpn uninstall` removing everything it asked about, rebooted and reinstalled, rebooted and restored my configuration backup, and rebooted a final time. That did not help. I then ran through all of the trouble shooting again and still cannot find why it will not route. 

**Expected behavior**

Once I have connected to WireGuard via a client connection I expect to be able to reach both the VPN network and have a route to the Internet. I have the former, but not the latter. 

**Screenshots**

### Have you searched for similar issues and solutions?
    (yes/no / which issues?)

yes

**Additional context**

### Have you taken any steps towards solving your issue?
```
  which?
```
Yes, as described in the problem description.

#### What type of hardware are you running PiVPN at?

```
Virtual machine -- AWS EC2 instance, t3a.small
```
#### Output of `uname -a`
```
Linux unifi.prime42.wtf 5.15.0-1019-aws #23-Ubuntu SMP Wed Aug 17 18:33:13 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```

#### Output of `cat /etc/os-release`
```
PRETTY_NAME="Ubuntu 22.04.1 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.1 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
```

### If install failed Please provide the console output of  `curl -L https://install.pivpn.io | bash`
```
INSTALL PASSED
```

### Console output of      `curl -L install.pivpn.io | bash`
```
We have detected an existing install.                              │ 
                      │ /etc/pivpn/wireguard/setupVars.conf                                │ 
                      │                                                                    │ 
                      │ Please choose from the following options (Reconfigure can be used  │ 
                      │ to add a second VPN type):                                         │ 
                      │                                                                    │ 
                      │        Update      Get the latest PiVPN scripts                    │ 
                      │        Repair      Reinstall PiVPN using existing settings         │ 
                      │        Reconfigure Reinstall PiVPN with new settings
```

### Console output of      `pivpn add` or `pivpn add nopass`
```
NOT USING OpenVPN
```

### Console output of      `pivpn debug`

```
::: Generating Debug Output
::::		PiVPN debug		 ::::
=============================================
::::		Latest commit		 ::::
Branch: master
Commit: e7d06d0545e1812b787aa0f31009df53b8741ef8
Author: 4s3ti
Date: Fri Aug 26 07:26:56 2022 +0200
Summary: docs(template): Update Issue template [skip ci]
=============================================
::::	    Installation settings    	 ::::
PLAT=Ubuntu
OSCN=jammy
USING_UFW=0
pivpnforceipv6route=1
IPv4dev=ens5
IPv4addr=10.103.42.174/20
IPv4gw=10.103.42.1
install_user=ubuntu
install_home=/home/ubuntu
VPN=wireguard
pivpnPORT=51280
pivpnDNS1=8.8.8.8
pivpnDNS2=8.8.4.4
pivpnHOST=REDACTED
INPUT_CHAIN_EDITED=0
FORWARD_CHAIN_EDITED=0
INPUT_CHAIN_EDITEDv6=
FORWARD_CHAIN_EDITEDv6=
pivpnPROTO=udp
pivpnMTU=1420
pivpnDEV=wg0
pivpnNET=10.225.241.0
subnetClass=24
pivpnenableipv6=0
ALLOWED_IPS="0.0.0.0/0, ::0/0"
UNATTUPG=1
INSTALLED_PACKAGES=(dnsutils grepcidr net-tools bsdmainutils iptables-persistent wireguard-tools qrencode)
=============================================
::::  Server configuration shown below   ::::
[Interface]
PrivateKey = server_priv
Address = 10.190.20.1/24
MTU = 1420
ListenPort = 51280
### begin John-Pixel4 ###
[Peer]
PublicKey = John-Pixel4_pub
PresharedKey = John-Pixel4_psk
AllowedIPs = 10.190.20.2/32
### end John-Pixel4 ###
### begin Casey ###
[Peer]
PublicKey = Casey_pub
PresharedKey = Casey_psk
AllowedIPs = 10.190.20.3/32
### end Casey ###
### begin john-laptop ###
[Peer]
PublicKey = john-laptop_pub
PresharedKey = john-laptop_psk
AllowedIPs = 10.190.20.4/32
### end john-laptop ###
=============================================
::::  Client configuration shown below   ::::
[Interface]
PrivateKey = John-Pixel4_priv
Address = 10.190.20.2/24
DNS = 8.8.8.8, 8.8.4.4

[Peer]
PublicKey = server_pub
PresharedKey = John-Pixel4_psk
Endpoint = unifi.prime42.wtf:51280
AllowedIPs = 0.0.0.0/0, ::0/0
=============================================
:::: 	Recursive list of files in	 ::::
::::	/etc/wireguard shown below	 ::::
/etc/wireguard:
configs
keys
wg0.conf

/etc/wireguard/configs:
Casey.conf
John-Pixel4.conf
clients.txt
john-laptop.conf

/etc/wireguard/keys:
Casey_priv
Casey_psk
Casey_pub
John-Pixel4_priv
John-Pixel4_psk
John-Pixel4_pub
john-laptop_priv
john-laptop_psk
john-laptop_pub
server_priv
server_pub
=============================================
::::		Self check		 ::::
:: [OK] IP forwarding is enabled
:: [OK] Iptables MASQUERADE rule set
:: [OK] WireGuard is running
:: [OK] WireGuard is enabled 
(it will automatically start on reboot)
:: [OK] WireGuard is listening on port 51280/udp
=============================================
:::: Having trouble connecting? Take a look at the FAQ:
:::: https://docs.pivpn.io/faq
=============================================
:::: WARNING: This script should have automatically masked sensitive       ::::
:::: information, however, still make sure that PrivateKey, PublicKey      ::::
:::: and PresharedKey are masked before reporting an issue. An example key ::::
:::: that you should NOT see in this log looks like this:                  ::::
:::: YIAoJVsdIeyvXfGGDDadHh6AxsMRymZTnnzZoAb9cxRe                          ::::
=============================================
::::		Debug complete		 ::::
::: 
::: Debug output completed above.
::: Copy saved to /tmp/debug.log
::: 
```

### IP Tables checks
```
ubuntu@unifi:~$ sudo iptables -S
-P INPUT ACCEPT
-P FORWARD ACCEPT
-P OUTPUT ACCEPT
ubuntu@unifi:~$ sudo iptables -t nat -S
-P PREROUTING ACCEPT
-P INPUT ACCEPT
-P OUTPUT ACCEPT
-P POSTROUTING ACCEPT
-A POSTROUTING -s 10.225.241.0/24 -o ens5 -m comment --comment wireguard-nat-rule -j MASQUERADE
```

### Short TCP Dump
```
ubuntu@unifi:~$ sudo tcpdump -n -i ens5 udp port 51280
tcpdump: verbose output suppressed, use -v[v]... for full protocol decode
listening on ens5, link-type EN10MB (Ethernet), snapshot length 262144 bytes
09:58:37.383051 IP 104.34.39.22.40659 > 10.103.42.174.51280: UDP, length 112
09:58:37.815206 IP 104.34.39.22.40659 > 10.103.42.174.51280: UDP, length 128
09:58:37.821402 IP 104.34.39.22.40659 > 10.103.42.174.51280: UDP, length 128
09:58:37.828746 IP 104.34.39.22.40659 > 10.103.42.174.51280: UDP, length 128
09:58:37.835753 IP 104.34.39.22.40659 > 10.103.42.174.51280: UDP, length 128
09:58:38.208240 IP 104.34.39.22.40659 > 10.103.42.174.51280: UDP, length 96
09:58:38.208241 IP 104.34.39.22.40659 > 10.103.42.174.51280: UDP, length 96
09:58:38.818848 IP 104.34.39.22.40659 > 10.103.42.174.51280: UDP, length 128
09:58:38.824772 IP 104.34.39.22.40659 > 10.103.42.174.51280: UDP, length 128
09:58:39.035167 IP 104.34.39.22.40659 > 10.103.42.174.51280: UDP, length 112
09:58:39.066176 IP 104.34.39.22.40659 > 10.103.42.174.51280: UDP, length 128
09:58:39.066176 IP 104.34.39.22.40659 > 10.103.42.174.51280: UDP, length 128
09:58:39.341787 IP 104.34.39.22.40659 > 10.103.42.174.51280: UDP, length 112
09:58:40.042034 IP 104.34.39.22.40659 > 10.103.42.174.51280: UDP, length 112
09:58:42.392762 IP 104.34.39.22.40659 > 10.103.42.174.51280: UDP, length 112
09:58:42.834883 IP 104.34.39.22.40659 > 10.103.42.174.51280: UDP, length 128
09:58:42.840769 IP 104.34.39.22.40659 > 10.103.42.174.51280: UDP, length 128
09:58:43.217563 IP 104.34.39.22.40659 > 10.103.42.174.51280: UDP, length 96
09:58:43.217563 IP 104.34.39.22.40659 > 10.103.42.174.51280: UDP, length 96
09:58:43.742943 IP 104.34.39.22.40659 > 10.103.42.174.51280: UDP, length 96
09:58:43.748112 IP 104.34.39.22.40659 > 10.103.42.174.51280: UDP, length 96
09:58:43.825922 IP 104.34.39.22.40659 > 10.103.42.174.51280: UDP, length 128
09:58:43.831217 IP 104.34.39.22.40659 > 10.103.42.174.51280: UDP, length 128
09:58:44.038034 IP 104.34.39.22.40659 > 10.103.42.174.51280: UDP, length 112
09:58:44.069685 IP 104.34.39.22.40659 > 10.103.42.174.51280: UDP, length 128
09:58:44.076068 IP 104.34.39.22.40659 > 10.103.42.174.51280: UDP, length 128
09:58:44.336849 IP 104.34.39.22.40659 > 10.103.42.174.51280: UDP, length 112
09:58:45.035380 IP 104.34.39.22.40659 > 10.103.42.174.51280: UDP, length 128
09:58:45.035380 IP 104.34.39.22.40659 > 10.103.42.174.51280: UDP, length 128
09:58:45.053517 IP 104.34.39.22.40659 > 10.103.42.174.51280: UDP, length 112
09:58:45.708833 IP 104.34.39.22.40659 > 10.103.42.174.51280: UDP, length 96
^C
31 packets captured
31 packets received by filter
0 packets dropped by kernel
```

I can run a ping to 10.190.20.1 and also to 8.8.8.8 (Google DNS) and let them run. The 10.190.20.1 ping will fail, and then start returning pongs as soon as I connect and then stop again when I disconnect. The 8.8.8.8 does the opposite, succeeding until I connect and then failing while connected, and then succeeding again as soon as disconnected. Also attempting to ping or traceroute while connected just immediately hangs with no output. 

The client side machine I did the testing on is a System76 Kudu 6 laptop:
```
NAME="Feren OS"
VERSION="2022.07"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Feren OS"
VERSION_ID="20.04"
HOME_URL="https://ferenos.weebly.com/"
SUPPORT_URL="https://ferenos.weebly.com/feren-os-on-social-media.html"
BUG_REPORT_URL="https://github.com/feren-OS/Bug-Reporting-Center/issues/new"
PRIVACY_POLICY_URL="https://ferenos.weebly.com/privacy-policy.html"
VERSION_CODENAME=focal
FEREN_CODENAME=krypton

```
However, I an having the same issue with an Android client on a Pixel 4. That is a huge issue as the Pixel 4 is setup to require allays on VPN or not allow internet access at all. 

Thank you in advance for your assistance.

**Comment by noel-schenk** [comment]
Same Issue with OpenVPN after upgrade to 22.04.

**Comment by bennybubble** [comment]
Same issue here, did anyone found a solution?
If I installed 22.04 from scratch, would this solve the issue?

**Reply by arfianadam** [reply]
No issues installing from scratch, tried it many times.

**Comment by T313C0mun1s7** [comment]
> No issues installing from scratch, tried it many times.

Yeah, you completely missed the point. Seems you failed to even read the title. The part where it says "after dist upgrade”

**Reply by arfianadam** [reply]
I was replying to @bennybubble , they asked if it would work if they installed pivpn after installing 22.04 from scratch.

**Reply by T313C0mun1s7** [reply]
Why would it not? That has nothing to do with this issue and only confuses the discourse. When issues don't stay on topic, they won't get solved.

**Comment by timboxyz** [comment]
I had a fully working version on Mint 21.2 (based on Jammy) and after an update to 21.3 (still based on Jammy) All I could connect to was the wireguard PC. (ping the tunnel address and the LAN IP address but nothing else on the LAN). This sounds like the same issue and one associated with routing tables or thereabouts. Repair install made no difference.

I have a Timeshift snapshot of before and after the update, so a chance of seeing what has changed, if I knew where to start looking.

**Reply by timboxyz** [reply]
Hmm, so on my system for some reason the MASQUERADE iptables rule vanished and was not replaced on a repair install. However, pivpn -d and saying yes to the prompt fixed it, and it survives reboots, so maybe not the same issue. But it does show it continues to work on Jammy.
